### DISCUSSION TITLE
Beatmap local leaderboards doesn't update when importing scores

### DISCUSSION THREAD

**Discussion by BlauFx** [ppy/osu#21948] [Q&A]
https://user-images.githubusercontent.com/34138240/210089038-e5bb9592-e0d2-4f0d-be5a-5d915f9b4ad6.mp4

[logs.zip](https://github.com/ppy/osu/files/10325416/logs.zip)

https://user-images.githubusercontent.com/34138240/210089090-94923583-a3a9-48b4-8deb-e47fcc4336d1.mp4

[logs import feature.zip](https://github.com/ppy/osu/files/10325417/logs.import.feature.zip)

As can be seen in the first video:
If I drag the replay into lazer (beatmap is not present) the replay won't be imported at all.
If I drag the replay into lazer (beatmap is present) but do not click on the notification to view it, I can't see my local scores until I click on the notification.

I have a few replays that I would like to import into lazer from stable however for some replays I do not have the required beatmap, just the replay. Lazer won't store the replays in the database, only if the beatmap is also present, is that correct? (see second video)

**Comment by frenzibyte** [comment]
> If I drag the replay into lazer (beatmap is not present) the replay won't be imported at all.

That's intended behaviour, the replay cannot be imported if the corresponding beatmap isn't in the database.

**Reply by BlauFx** [reply]
That's unfortunate. Couldn't lazer store the replay in the database and at some point if the the corresponding beatmap is imported then link the replay to the beatmap afterwards and process it? With this approach it's possible to import all replays from stable without worrying what's imported or what's not.

**Comment by peppy** [comment]
I have no idea what's even going on in the first video. The local leaderboard should not enter a loading state like that.
